packages/agents/src/lib/Agent.ts

```diff
@@ -33,11 +33,11 @@ export class Agent implements AgentInterface {
   name = ''
   id: any
   secrets: any
-  publicVariables: Record<string, string>
-  data: AgentInterface
+  publicVariables!: Record<string, string>
```

Comment:
They tell Typescript to not worry that these properties may not be defined directly in the constructor but they aren't undefined. Since I moved setting those properties into the update function the constructor calls we needed to tell typescript not to worry they aren't defined in the constructor itself.